spec/renuo/bin-check/dsl_config_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'spec_helper'
+require './lib/renuo_bin_check/dsl_config'
+
+RSpec.describe DSLConfig do
+  it 'gets the right infos out of a new check' do
+    dsl_config = DSLConfig.new('rspec') do
+      files 'lib/**/*.rb'
+      command 'rubocop'
+      reversed_exit false
+    end
+
+    expect(dsl_config.configs).to eq(name: 'rspec', files: 'lib/**/*.rb', command: 'rubocop', reversed_exit: false)
+    expect(dsl_config.children?).to eq(false)
+  end
+
+  it 'creates new children if there are nested checks' do
+    dsl_config = DSLConfig.new('ruby_files') do
+      files 'lib/**/*.rb'
+      rubocop do
+        command 'rubocop'
+        reversed_exit false
+      end
+      error_message 'an error occured'
+      p_finder do
+        command 'find p'
+        reversed_exit true
+      end
+    end
+
+    expect(dsl_config.configs).to eq(name: 'ruby_files', files: 'lib/**/*.rb', error_message: 'an error occured')
+    expect(dsl_config.children.first.configs).to eq(name: 'rubocop', command: 'rubocop', reversed_exit: false)
```

Comment:
i was asking myself while reading this test, if the config should already include 'files'  of its parent because otherwise it has to know the parents item. but you solve that in converting it to a config-model afterwards (maybe in a service) right?